ros/processor/report_processor_event_producer.py

```diff
@@ -15,14 +28,142 @@ def api_and_no_pcp_raw_payload(payload):
+        "component": TELEMETRY_COMPONENT,
+        "system_id": system_id
+    }]
+
+
+def _determine_candidates(report_output, error_key):
+    """Determine top candidate and price with proper fallbacks."""
+    if error_key in NO_RECOMMENDATION_STATES:
+        return None, None
+
+    candidates = report_output.get("candidates", [[None, None]])
+    if not candidates:
+        return None, None
+
+    top_candidate = candidates[0][0] if candidates[0] else None
+    top_candidate_price = candidates[0][1] if candidates[0] else None
+    return top_candidate, top_candidate_price
+
+
+def _calculate_recommendations_count(error_key):
+    """Calculate number of recommendations using process_report logic."""
+    return -1 if error_key == 'NO_PCP_DATA' else 1
+
+
+def _prepare_performance_record(report_perf_profile):
+    """Prepare performance record by removing unnecessary keys."""
+    # Remove the 'type' key as it's not required
+    report_perf_profile.pop('type', None)
+    return report_perf_profile
+
+
+def perf_profile_and_rule_payload(report_metadata_output, report_output, report_perf_profile):
+    """Build performance profile and rules payload for event publishing.
+
+    Args:
+        report_metadata_output: Metadata from report processing
+        report_output: Main report output with states and recommendations
+        report_perf_profile: Performance profile data
+
+    Returns:
+        dict: Complete payload for event publishing
+    """
+    # Extract basic data
+    error_key = report_output.get("error_key")
+    state = SystemStatesWithKeys[error_key].value if error_key else None
+    system_states = _extract_system_states(report_output)
+
+    # Process performance data
+    performance_record = _prepare_performance_record(report_perf_profile)
+    performance_utilization = _calculate_performance_utilization(report_metadata_output)
+
+    # Build recommendations and rule details
+    rule_hit_details = _build_rule_hit_details(error_key, report_output, report_metadata_output.get("system_id"))
+    number_of_recommendations = _calculate_recommendations_count(error_key)
+    top_candidate, top_candidate_price = _determine_candidates(report_output, error_key)
+
+    # Assemble final payload
+    return {
+        **system_states,  # cpu_states, io_states, memory_states
+        "state": state,
+        "instance_type": report_output.get("instance_type"),
+        "region": report_output.get("region"),
+        "performance_record": performance_record,
+        "performance_utilization": performance_utilization,
+        "report_date": datetime.now(timezone.utc).isoformat(),
+        "rule_hit_details": rule_hit_details,
+        "number_of_recommendations": number_of_recommendations,
+        "psi_enabled": report_metadata_output.get("psi_enabled"),
+        "top_candidate": top_candidate,
+        "top_candidate_price": top_candidate_price
+    }
+
+
+def produce_report_processor_event_pcp_raw_data(
```

Comment:
Can we use the `produce_report_processor_event` method with some modification? Both methods are the same. Here, we have to add `perf_profile_and_rule`.